haskell/haddock.bzl

```diff
@@ -25,6 +25,8 @@ def _get_haddock_path(package_id):
 def _haskell_doc_aspect_impl(target, ctx):
     if HaskellBuildInfo not in target or HaskellLibraryInfo not in target:
         return []
+    if HaddockInfo in target:
+        return target[HaddockInfo]
```

Comment:
The `haskell_import` rule also returns an `HaddockInfo` provider (because the haddocks are already here in the imported package) which is the reason for this change. I'll add a comment to explain it